config/internal/apiserver/monitor.yaml.tmpl

```
@@ -0,0 +1,16 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    app: ds-pipeline-{{.Name}}
```

```suggestion
    app: ds-pipeline-servicemonitor-{{.Name}}
```
as we having different naming, lets follow the same pattern.

---

I meant on labels.app itself, not on the other parts those looks good to me
anyhow this is a hard requirement, please ignore it.

---

oh. Huh. I didn't notice that the labels.app is changing all over the code. I think I misunderstood what that was for.

---

I kept the labels.app matching to what we have in the service and route:
https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/internal/apiserver/route.yaml.tmpl#L7
https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/internal/apiserver/service.yaml.tmpl#L9

The matchlabels in ServiceMonitor are also exactly the same as label.app in the service. My suggestion would be to keep the labels.app as is, consistent with the service and route. What do you think @harshad16 @gregsheremeta ?

---

I hesitate to comment further because I don't know what it does :joy:

---

Technically `matchlabels` in `selector` section is the only important bit as per fetching metrics from apisever which absolutely correct written.
i was just suggesting on the labels, as labels in metadata are used for filtering.
anyhow this is good for me, no changes needed here
thanks @DharmitD